geopandas/explore.py

```diff
@@ -337,6 +337,9 @@ def _colormap_helper(_cmap, n_resample=None, idx=None):
     if len(json_not_supported_cols) > 0:
         gdf = gdf.astype({c: "string" for c in json_not_supported_cols})
 
+    if is_datetime64_any_dtype(gdf.index):
```

Comment:
don't know what happened but I swear I left a comment on this. We should use the same logic as above, so that we can't run into other object dtype non serialisable issues in the index. I'll aim to do a follow up.